Assets/Scripts/TaskGenerator.cs

```diff
@@ -0,0 +1,156 @@
+using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TaskGenerator : MonoBehaviour
+{
+    private Dictionary<string, GameObject[]> allInteractableObjects;
+    public GameObject[] GamePrefabs;
+    public MiniGameManager miniGameManager;
+
+    private List<GameObject> gameObjectsWithTasks;
+
+    //Materials, will remove
+    public Material canSolveMaterial;
+    public Material fixedMaterial;
+    public Material failedMaterial;
+
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        SetUpAllData();
+        SetUpEvents();
+
+        ChooseAllTasksAtStart();
+    }
+
+    private void SetUpAllData()
+    {
+        gameObjectsWithTasks = new List<GameObject>();
+        allInteractableObjects = new Dictionary<string, GameObject[]>();
+
+        allInteractableObjects.Add("StreetLamp", GameObject.FindGameObjectsWithTag("StreetLamp"));
+        allInteractableObjects.Add("ManHole", GameObject.FindGameObjectsWithTag("ManHole"));
+        allInteractableObjects.Add("Tree", GameObject.FindGameObjectsWithTag("Tree"));
+        allInteractableObjects.Add("SolarPanel", GameObject.FindGameObjectsWithTag("SolarPanel"));
+    }
+
+    private void SetUpEvents()
+    {
+        miniGameManager.OnGameOver += MiniGameManager_OnGameOver;
+        miniGameManager.OnGameWon += MiniGameManager_OnGameWon;
+    }
+
+    private void AddTasksToObject(List<GameObject> allObjects, string objectType)
+    {
+        System.Random random = new System.Random();
+
+        foreach(GameObject gameObject in allObjects)
+        {
+            List<GameObject> perfabs;
+            GameObject gamePrefab;
+
+            //Decided which game to display for each object
+            switch (objectType)
+            {
+                case "StreetLamp":
+                    perfabs = GamePrefabs.Where(x => x.name.Contains("Rewire") || x.name.Contains("ColorBeep")).ToList();
+                    gamePrefab = perfabs[random.Next(perfabs.Count)];
+                    break;
+                case "ManHole":
+                    perfabs = GamePrefabs.Where(x => x.name.Contains("Sewage")).ToList();
+                    gamePrefab = perfabs[random.Next(perfabs.Count)];
+                    break;
+                case "Tree":
+                    perfabs = GamePrefabs.Where(x => x.name.Contains("Dig")).ToList();
+                    gamePrefab = perfabs[random.Next(perfabs.Count)];
+                    break;
+                case "SolarPanel":
+                    perfabs = GamePrefabs.Where(x => x.name.Contains("Rewire") || x.name.Contains("ColorBeep")).ToList();
+                    gamePrefab = perfabs[random.Next(perfabs.Count)];
+                    break;
+                default:
+                    gamePrefab = GamePrefabs[random.Next(GamePrefabs.Length)];
+                    break;
```

Comment:
Unfortunately this is really not that scalable and easy to work with, especially the Linq part (the .Contains is bound to break at some point when I change the name and that would be a hard bug to find out)

Reply:
I really tried to see what I could do for the .Contains , but don't know how to do it otherwise. Do you have advice?